### PR TITLE
feat: enable XFS online checks and repair in kernel

### PR DESCRIPTION
**Pull request by dsseng**
Signed-off-by: Dmitry Sharshakov <dmitry.sharshakov@siderolabs.com>

### CODE REVIEW

File: kernel/build/config-amd64 [hunk 1]
```diff
@@ -17,7 +17,6 @@ CONFIG_CC_CAN_LINK=y
 CONFIG_CC_CAN_LINK_STATIC=y
 CONFIG_CC_HAS_ASM_GOTO_OUTPUT=y
 CONFIG_CC_HAS_ASM_GOTO_TIED_OUTPUT=y
-CONFIG_TOOLS_SUPPORT_RELR=y
```
Comment (frezbo) [inline kernel/build/config-amd64:20]: needs to run `old-defconfig`

Comment (dsseng) [reply kernel/build/config-amd64:20]: Am I missing something?
```
WARNING: The requested image's platform (linux/arm64) does not match the detected host platform (linux/amd64/v3) and no specific platform was requested
  HOSTCC  scripts/basic/fixdep
make[2]: *** No rule to make target 'old-defconfig'.  Stop.
make[1]: *** [/src/Makefile:679: old-defconfig] Error 2
```

Comment (dsseng) [reply kernel/build/config-amd64:20]: nvm it's kernel-olddefconfig

**Pull request merged**